config/initializers/sidekiq.rb

```diff
@@ -1,4 +1,9 @@
-redis = { namespace: Rails.env, url: "redis://localhost:6379/#{ENV['REDIS_DB_NUMBER'] || 0}" }
+redis = { namespace: Rails.env, url: "redis://localhost:6379/#{ENV['SIDEKIQ_DB_NUMBER'] || 0}" }
+
+if Rails.env == "test"
+  redis = { namespace: Rails.env, url: "redis://localhost:6379/0" }
```

Comment:
Check for [hard spaces](https://en.wikipedia.org/wiki/Hard_space), probably after the first `{`.
I write them all the time.